test/spec/writer/operations.zig

```diff
@@ -0,0 +1,96 @@
+const std = @import("std");
+const spec_test_options = @import("spec_test_options");
+const ForkSeq = @import("config").ForkSeq;
+const Preset = @import("preset").Preset;
+const Operation = @import("../test_type/operations.zig").Operation;
+
+pub const header =
+    \\// This file is generated by write_spec_tests.zig.
+    \\// Do not commit changes by hand.
+    \\
+    \\const std = @import("std");
+    \\const ForkSeq = @import("config").ForkSeq;
+    \\const preset = @import("preset").preset.preset;
+    \\const spec_test_options = @import("spec_test_options");
+    \\const Operation = @import("../test_type/operations.zig").Operation;
+    \\
+    \\const allocator = std.heap.page_allocator;
+    \\
+    \\
+;
+
+const test_template =
+    \\test "{s} {s} {s}" {{
+    \\    const test_dir_name = try std.fs.path.join(allocator, &[_][]const u8{{
+    \\        spec_test_options.spec_test_out_dir,
+    \\        spec_test_options.spec_test_version,
+    \\        @tagName(preset) ++ "/tests/" ++ @tagName(preset) ++ "/{s}/operations/{s}/pyspec_tests/{s}",
+    \\    }});
+    \\    defer allocator.free(test_dir_name);
+    \\
+    \\    const test_dir = std.fs.cwd().openDir(test_dir_name, .{{}}) catch return error.SkipZigTest;
+    \\    var tc = try Operation.TestCase(.{s}, .{s}, {}).init(allocator, test_dir);
+    \\    defer tc.deinit();
+    \\
+    \\    try tc.runTest();
+    \\}}
+    \\
+    \\
+;
+
+pub fn writeOperationsTests(comptime forks: []const ForkSeq, writer: std.io.AnyWriter) !void {
```

Comment:
I think this file reading logic will be reused so many times for other runners. 
Better write a generic version of it.

Reply:
ok take a look now (at `write_spec_tests.zig`). More of a framework for handling different runners and handlers and a unified directory walker there.